client/relay-chain-network/src/lib.rs

```diff
@@ -0,0 +1,460 @@
+		let response_bytes = self
+			.call_remote_runtime_function(
+				"ParachainHost_candidate_pending_availability",
+				at,
+				Some(para_id.encode()),
+			)
+			.await?;
+
+		Ok(Option::<CommittedCandidateReceipt<PHash>>::decode(&mut &*response_bytes.0)?)
+	}
+
+	async fn parachain_host_session_index_for_child(
+		&self,
+		at: &PHash,
+	) -> Result<SessionIndex, RelayChainError> {
+		let response_bytes = self
+			.call_remote_runtime_function("ParachainHost_session_index_for_child", at, None)
+			.await?;
+
+		Ok(SessionIndex::decode(&mut &*response_bytes.0)?)
+	}
+
+	async fn parachain_host_validators(
+		&self,
+		at: &PHash,
+	) -> Result<Vec<ValidatorId>, RelayChainError> {
+		let response_bytes =
+			self.call_remote_runtime_function("ParachainHost_validators", at, None).await?;
+
+		Ok(Vec::<ValidatorId>::decode(&mut &*response_bytes.0)?)
+	}
+
+	async fn parachain_host_persisted_validation_data(
+		&self,
+		block_id: &PHash,
+		para_id: ParaId,
+		occupied_core_assumption: OccupiedCoreAssumption,
+	) -> Result<Option<PersistedValidationData>, RelayChainError> {
+		let response_bytes = self
+			.call_remote_runtime_function(
+				"ParachainHost_persisted_validation_data",
+				block_id,
+				Some((para_id, occupied_core_assumption).encode()),
+			)
+			.await?;
+
+		Ok(Option::<PersistedValidationData>::decode(&mut &*response_bytes.0)?)
+	}
+
+	async fn parachain_host_inbound_hrmp_channels_contents(
+		&self,
+		para_id: ParaId,
+		at: &PHash,
+	) -> Result<BTreeMap<ParaId, Vec<InboundHrmpMessage>>, RelayChainError> {
+		let response_bytes = self
+			.call_remote_runtime_function(
+				"ParachainHost_inbound_hrmp_channels_contents",
+				at,
+				Some(para_id.encode()),
+			)
+			.await?;
+
+		Ok(BTreeMap::<ParaId, Vec<InboundHrmpMessage>>::decode(&mut &*response_bytes.0)?)
+	}
+
+	async fn parachain_host_dmq_contents(
+		&self,
+		para_id: ParaId,
+		at: &PHash,
+	) -> Result<Vec<InboundDownwardMessage>, RelayChainError> {
+		let response_bytes = self
+			.call_remote_runtime_function("ParachainHost_dmq_contents", at, Some(para_id.encode()))
+			.await?;
+
+		Ok(Vec::<InboundDownwardMessage>::decode(&mut &*response_bytes.0)?)
+	}
+
+	async fn subscribe_all_heads(&self) -> Result<Subscription<PHeader>, RelayChainError> {
+		self.subscribe::<PHeader>("chain_subscribeAllHeads", "chain_unsubscribeAllHeads", None)
+			.await
+	}
+
+	async fn subscribe_new_best_heads(&self) -> Result<Subscription<PHeader>, RelayChainError> {
+		self.subscribe::<PHeader>("chain_subscribeNewHeads", "chain_unsubscribeNewHeads", None)
+			.await
+	}
+
+	async fn subscribe_finalized_heads(&self) -> Result<Subscription<PHeader>, RelayChainError> {
+		self.subscribe::<PHeader>(
+			"chain_subscribeFinalizedHeads",
+			"chain_unsubscribeFinalizedHeads",
+			None,
+		)
+		.await
+	}
+}
+
+/// RelayChainNetwork is used to interact with a full node that is running locally
+/// in the same process.
+#[derive(Clone)]
+pub struct RelayChainNetwork {
```

Comment:
The relay chain network is a bad name, because you normally call all relay chain networks the "relay chain network"